src/DotNetBoilerplate.Core/Catalogs/ICatalogRepository.cs

```diff
@@ -0,0 +1,28 @@
+﻿using DotNetBoilerplate.Core.Books;
+
+namespace DotNetBoilerplate.Core.Catalogs;
+
+public interface ICatalogRepository
+{
+    Task AddAsync(Catalog catalog);
+
+    Task<bool> UserCanNotAddCatalogAsync(Guid bookStoreId);
```

Comment:
Te nazwy tutaj też niewiele mówią - nie wiadomo co to sprawdza póki się nie wejdzie do implementacji na bazie